receiver/prometheusreceiver/internal/metricsbuilder.go

```diff
@@ -87,6 +88,23 @@ func (b *metricBuilder) matchStartTimeMetric(metricName string) bool {
 
 // AddDataPoint is for feeding prometheus data complexValue in its processing order
 func (b *metricBuilder) AddDataPoint(ls labels.Labels, t int64, v float64) error {
+	// Any datapoint with duplicate labels MUST be rejected per:
+	// * https://github.com/open-telemetry/wg-prometheus/issues/44
+	// * https://github.com/open-telemetry/opentelemetry-collector/issues/3407
+	// as Prometheus rejects such too as of version 2.16.0, released on 2020-02-13.
+	seen := make(map[string]bool)
+	dupLabels := make([]string, 0, len(ls))
+	for _, label := range ls {
+		if _, ok := seen[label.Name]; ok {
+			dupLabels = append(dupLabels, label.Name)
+		}
+		seen[label.Name] = true
+	}
+	if len(dupLabels) != 0 {
+		sort.Strings(dupLabels)
+		return fmt.Errorf("invalid sample: non-unique label names: %q", dupLabels)
```

Comment:
nit: it might be nice to seek the label keys and values, instead of just the name for debugging.

Reply:
Good idea, I debated that too, but Prometheus doesn't print out the values though hence just tried to mirror what they do; Also printing out the Name:Value after printing out a list of keys, might make for huge errors. The Label values might be long, which sure would require newlines, but lots of rabbit holes to go into. Perhaps we can consider this in the future if someone asks for it?

Reply:
Works for me.  Thanks for investigating.

Reply:
Thanks for raising it!